common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/SourceIdentifier.scala

```diff
@@ -6,5 +6,7 @@ case class SourceIdentifier(
   ontologyType: String,
   value: String
 ) {
+  assert(!value.endsWith(" "))
```

Comment:
Couple of notes:

* Right now this will throw a generic Assertion failed message; I'd add a second argument that logs the 

    ```suggestion
      assert(..., s"Value contains trailing whitespace: <$value>")
    ```

* I would make the condition `value == value.trim`, so we handle all trailing whitespace (not just space, but tab/newline) and also leading whitespace.

* Minor: I prefer `require` to `assert` for this sort of thing. From the [Prede(f docs](https://www.scala-lang.org/api/current/scala/Predef$.html#require(requirement:Boolean,message:=>Any):Unit)

    > `require` is distinct from `assert` in that if the condition fails, then the caller of the function is to blame rather than a logical error having been made within [the function] itself.

    But that's just a preference, and I'm not going to fuss if you leave it as an assert.